packages/xsnap/src/xsnap.js

```diff
@@ -73,24 +79,32 @@ export function xsnap(options) {
   const vatExit = defer();
 
   const args = snapshot ? ['-r', snapshot] : [];
+  if (meteringLimit !== undefined) {
+    args.push('-l', `${meteringLimit}`);
+  }
 
   const xsnapProcess = spawn(bin, args, {
     stdio: ['ignore', stdout, stderr, 'pipe', 'pipe'],
   });
 
+  let meterStats;
```

Comment:
I'm not happy with this local state to communicate the meter usage to the `vatCancelled` promise.  It's useful functionality, but ugly.

Any suggestions on how to make this better would be appreciated.

Reply:
Perhaps @kriskowal would care to weigh in?

There's a lot going on with async and generators and stuff here. I think I had it all swapped in at one time, but I don't now, and this variable is sufficiently local that I'm OK with it.

I'd prefer to see a type annotation.

Reply:
I followed @warner's recommendation only to propagate these stats when returning OK (`.`).  It did indeed clear up some of the convolutions.